src/transformations.jl

```diff
@@ -8,16 +8,16 @@ abstract type AbstractDataTransform end
 
 Apply transformation `t` to vector or matrix `x` in place.
 """
-transform!(t::AbstractDataTransform, x::AbstractArray{<:Real,1}) = transform!(x, t, x)
-transform!(t::AbstractDataTransform, x::AbstractArray{<:Real,2}) = transform!(x, t, x)
+transform!(t::AbstractDataTransform, x::AbstractMatrix{<:Real}; dims::Int=1) = transform!(x, t, x, Val(dims))
+transform!(t::AbstractDataTransform, x::AbstractVector{<:Real}) = transform!(x, t, x)
```

Comment:
Should also accept `dims`, and check that it's equal to `1`.

Reply:
OK

Reply:
Hummmmm........It seems redundant to a vector, because the `dims` has to be `1`, further with assertion check.

Reply:
Yes, but that way you can pass `dims=1` and it works, whether `x` is a vector or a matrix. Also, I'm not completely sure it's redundant, since it could make sense to allow treating vectors like single-column matrices in the future, allowing `dims=2`: for example, `sum([1, 2], dims=2)` is allowed is the identity.

Reply:
Oh, Cool!